src/mqtt.py

```diff
@@ -57,6 +57,7 @@ def connect(self):
 
     def disconnect(self):
         self._client.disconnect()
+        self._queue.put(None)
```

Comment:
```
     def disconnect(self):
          self._client.disconnect()		          
          self._client.disconnect()
          self._queue.put(None)
```
Why do you need to put None in queue?
It will be skipped by handling loop.